sklearn/tests/test_calibration.py

```diff
@@ -341,3 +348,31 @@ def decision_function(self, X):
     calibrated_clf = CalibratedClassifierCV(MockTensorClassifier())
     # we should be able to fit this classifier with no error
     calibrated_clf.fit(X, y)
+
+
+def test_calibration_pipeline():
+    # Test that calibration works in pre-fit pipeline with transformer,
+    # where X is not array-like, sparse matrix or dataframe at the start. See
+    # issue #8710
+    fake_features = [
+        {'state': 'NY', 'age': 'adult'},
+        {'state': 'TX', 'age': 'adult'},
+        {'state': 'VT', 'age': 'child'},
+    ]
+    labels = [1, 0, 1]
+
+    pipeline = Pipeline([
+                ('vectorizer', DictVectorizer()),
+                ('clf', RandomForestClassifier())
+        ])
+    pipeline.fit(fake_features, labels)
+
+    calib_clf = CalibratedClassifierCV(pipeline, cv='prefit')
+    calib_clf.fit(fake_features, labels)
+
+    # Check attributes are obtained from fitted estimator
+    assert_array_equal(calib_clf.classes_, [0, 1])
+    # `DictVectorizer` has no `n_features_in_`
+    msg = "'CalibratedClassifierCV' object has no attribute 'n_features_in_'"
```

Comment:
I would put this in another test to check `n_features_in_` behaviour. Basically it should give `n_features_in_` of the first estimator of a pipeline or raise an error with weird estimator and should be computed with `cv != "prefit"`